tests/components/suez_water/conftest.py

```diff
@@ -13,3 +35,39 @@ def mock_setup_entry() -> Generator[AsyncMock]:
         "homeassistant.components.suez_water.async_setup_entry", return_value=True
     ) as mock_setup_entry:
         yield mock_setup_entry
+
+
+@pytest.fixture(name="suez_client")
+def mock_suez_client(hass: HomeAssistant) -> Generator[AsyncMock]:
+    """Create mock for suez_water external api."""
+
+    mock = AsyncMock(spec=SuezClient)
```

Comment:
I suggest patching the library like we do in `smarty`. There we create 1 mock for both the config flow and the normal usage

Reply:
that's what I tought I had done :sweat: